### PR TITLE
http2: set the correct URL in pushed transfers

### LINKED ISSUE #5589: curl_easy_getinfo CURLINFO_EFFECTIVE_URL is the same for all handles created in the PUSHFUNCTION callback
When using libcurl multi handle with the PUSHFUNCTION option set to allow PUSH_PROMISE frames, if you call `curl_get_info(handle, CURLINFO_EFFECTIVE_URL, &result)` on one of the handles libcurl created to transfer the new data pushed by the server it returns the URL of the original request.

I don't know if this is the intended behavior or a possible bug, so I'm opening this issue here.

If this is in fact the intended behavior, a new info option to retrieve the pushed data URL would be very helpful, or at least some kind of flag saying this easy handle was created to transfer data pushed by the server.

### I did this
Ran the example at https://curl.haxx.se/libcurl/c/http2-pushinmemory.html with some minor edits:
1. Changed URL to https://http2.golang.org/serverpush
2. Allowed all pushes on `CURLMOPT_PUSHFUNCTION` callback
3. Called `curl_easy_getinfo(e, CURLINFO_EFFECTIVE_URL, &result);` before `curl_easy_cleanup(e);` in the `do/while` loop.

It shows the same URL for all requests.

### I expected the following

Different URLs for each easy handle. 

### curl/libcurl version
```
Version: libcurl/7.69.1-DEV OpenSSL/1.1.1d zlib/1.2.11 WinIDN libssh2/1.9.0_DEV nghttp2/1.40.0
Protocols: dict, file, ftp, ftps, gopher, http, https, imap, imaps, ldap, ldaps, pop3, pop3s, rtsp, scp, sftp, smb, smbs, smtp, smtps, telnet, tftp
Features: AsynchDNS, IDN, IPv6, Largefile, SSPI, Kerberos, SPNEGO, NTLM, SSL, libz, HTTP2, HTTPS-proxy
```

### operating system
Windows 10 x64

### PR DESCRIPTION
**Pull request by bagder**
...previously CURLINFO_EFFECTIVE_URL would report the URL of the
original "mother transfer", not the actually pushed resource.

Reported-by: Jonathan Cardoso Machado
Fixes #5589

### CODE REVIEW

File: lib/http2.c [hunk 1]
```diff
@@ -505,6 +545,10 @@ static int push_promise(struct Curl_easy *data,
       goto fail;
     }
 
+    rv = set_transfer_url(data, &heads);
```
Comment (JCMais) [inline lib/http2.c:548]: `data` here is the parent handle, did you meant to use `newhandle` instead?

**Comment by JCMais** [comment]
Thanks for starting the work on that so fast @bagder!

File: lib/http2.c (hunk 1 above)
Comment (bagder) [reply lib/http2.c:548]: oops, yes I did!

**Comment by bagder** [comment]
I used this code to reproduce and verify my fix: [5589.c](https://github.com/curl/curl/files/4819669/5589.c.txt)

**Comment by JCMais** [comment]
@bagder will this make into `7.71` or I should expect it to be available only with `7.72`?

**Comment by bagder** [comment]
Yeah, its really a question of how "dangerous" this fix is...  I think I prefer to play it safe as the release is already in twelve hours, so I I'll play this one safe and aim to get this merged in 7.72 instead.

**Comment by bagder** [comment]
@JCMais, did you get a chance to try out the patch for your use case?

**Comment by JCMais** [comment]
> @JCMais, did you get a chance to try out the patch for your use case?

Not yet, but I should be able tomorrow. I will report back.

**Comment by JCMais** [comment]
@bagder I've just tested your patch and it works fine

**Comment by bagder** [comment]
thanks for the confirmation @JCMais, merged now!

**Pull request closed without merging**